lib/Backend/Opnd.cpp

```diff
@@ -1777,6 +1784,73 @@ FloatConstOpnd::FreeInternal(Func *func)
 
 ///----------------------------------------------------------------------------
 ///
+/// Float32ConstOpnd::New
+///
+///     Creates a new Float32ConstOpnd.
+///
+///----------------------------------------------------------------------------
+
+Float32ConstOpnd *
+Float32ConstOpnd::New(float value, IRType type, Func *func)
+{
+    Assert(type == IRType::TyFloat32); //TODO: should we even allow specifying a type here? It should always be TyFloat32
+    Float32ConstOpnd * Float32ConstOpnd;
+
+    Float32ConstOpnd = JitAnew(func->m_alloc, IR::Float32ConstOpnd);
+
+    Float32ConstOpnd->m_value = value;
+    Float32ConstOpnd->m_type = type;
+    Float32ConstOpnd->m_kind = OpndKindFloat32Const;
+
+    return Float32ConstOpnd;
+}
+
+///----------------------------------------------------------------------------
+///
+/// Float32ConstOpnd::Copy
+///
+///     Returns a copy of this opnd.
+///
+///----------------------------------------------------------------------------
+
+Float32ConstOpnd *
+Float32ConstOpnd::CopyInternal(Func *func)
+{
+    Assert(m_kind == OpndKindFloat32Const);
+    Float32ConstOpnd * newOpnd;
+
+    newOpnd = Float32ConstOpnd::New(m_value, m_type, func);
```

Comment:
That means we would have to write 
```cpp
JitAnew(func->m_alloc, IR::Float32ConstOpnd, m_value, m_type);
```` 
Do we really want to go down that route ? Or am I missing something ?